app/assets/javascripts/mirador_widget_admin.js

```diff
@@ -156,6 +152,18 @@
         block.find('[name="mirador_config"]').replaceWith(_.template(template));
       },
 
+      addItemToSection: function(block, itemObject, shouldTriggerEvent) {
+        var index = block.find('input[type="hidden"][data-behavior="mirador-item"]').length;
+
+        itemsSection(block).append(
+          MiradorWidgetBlock.hiddenInput(index, itemObject)
+        );
+
+        if (shouldTriggerEvent || shouldTriggerEvent === undefined) {
```

Comment:
@jkeck Would you mind throwing a comment in here? It took me a minute to grok, but if I'm reading this correctly, this is only triggered for `CreateItemPanel`, right?

Reply:
Comment added, let me know if that looks good.